provider/pkg/provider/provider.go

```diff
@@ -2744,17 +2744,27 @@ func shouldNormalize(uns *unstructured.Unstructured) bool {
 	return kinds.KnownGroupVersions.Has(uns.GetAPIVersion())
 }
 
-// normalize converts an Unstructured resource into a normalized form so that semantically equivalent representations
+// normalizeInputs converts an Unstructured resource into a normalized form so that semantically equivalent representations
 // are set to the same output shape. This is important to avoid generating diffs for inputs that will produce the same
 // result on the cluster.
-func normalize(uns *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+func normalizeInputs(uns *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 	if shouldNormalize(uns) {
 		normalized, err := clients.Normalize(uns)
 		if err != nil {
 			return nil, err
 		}
 		uns = pruneLiveState(normalized, uns)
 	}
+
+	// Remove read-only fields
+	unstructured.RemoveNestedField(uns.Object, "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "deletionGracePeriodSeconds")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "deletionTimestamp")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "generation")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "managedFields")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "resourceVersion")
+	unstructured.RemoveNestedField(uns.Object, "metadata", "uid")
```

Comment:
This list of fields is from the [Kubernetes API Reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#objectmeta-v1-meta).

Reply:
Can we add this link into the codebase as a comment for future reference as well?